classes/controllers/tab/workflow/PKPVersioningTabHandler.inc.php

```diff
@@ -0,0 +1,204 @@
+<?php
+
+/**
+ * @file controllers/tab/workflow/PKPVersioningTabHandler.inc.php
+ *
+ * Copyright (c) 2014-2016 Simon Fraser University Library
+ * Copyright (c) 2003-2016 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class ReviewRoundTabHandler
+ * @ingroup controllers_tab_workflow
+ *
+ * @brief Handle AJAX operations for version tabs on production stages workflow pages.
+ */
+
+// Import the base Handler.
+import('classes.handler.Handler');
+import('lib.pkp.classes.core.JSONMessage');
+
+class PKPVersioningTabHandler extends Handler {
+
+	/**
+	 * Constructor
+	 */
+	function __construct() {
+		parent::__construct();
+	}
+
+
+	//
+	// Extended methods from Handler
+	//
+	/**
+	 * @copydoc PKPHandler::authorize()
+	 */
+	function authorize($request, &$args, $roleAssignments) {
+		// We need a submission revision id in request.
+		import('lib.pkp.classes.security.authorization.internal.VersioningRequiredPolicy');
+		$this->addPolicy(new VersioningRequiredPolicy($request, $args));
+
+		return parent::authorize($request, $args, $roleAssignments);
+	}
+
+	/**
+	 * create new submission revision
+	 * @param $args array
+	 * @param $request PKPRequest
+	 * @return JSONMessage JSON object
+	 */
+	function newVersion($args, $request){
+		$submissionId = (int)$request->getUserVar('submissionId');
+		$submissionDao = Application::getSubmissionDAO();
+		$submissionRevision = $submissionDao->getLatestRevisionId($submissionId);
+		// get data of the old version
+		$submission = $submissionDao->getById($submissionId, null, false, $submissionRevision);
+		// authors
+		$authorDao = DAORegistry::getDAO('AuthorDAO');
+		$authors = $authorDao->getBySubmissionId($submissionId, true, false, $submissionRevision);
+		// galleys
+		import('classes.article.ArticleGalley');
+		$articleGalleyDao = DAORegistry::getDAO('ArticleGalleyDAO');
```

Comment:
article galley should not be in pkp-lib. you could take a look at all the changes here and move everything that does not belong to OMP as well to OJS, but you can assume that the versioning will come for OMP soon too, thus leave the common things...